x-pack/filebeat/processors/decode_cef/cef/cef.rl

```diff
@@ -124,13 +124,13 @@ func (e *Event) unpack(data string) error {
         extension_key_start_chars = alnum | '_';
         extension_key_chars = extension_key_start_chars | '.' | ',' | '[' | ']';
         extension_key_pattern = extension_key_start_chars extension_key_chars*;
-        extension_value_chars = backslash | escape_equal | (any -- equal -- escape);
+        extension_value_chars_nospace = backslash | escape_equal | (any -- equal -- escape -- space);
 
         # Extension fields.
         extension_key = extension_key_pattern >mark %extension_key;
-        extension_value = (extension_value_chars @extension_value_mark)* >extension_value_start $err(extension_err);
-        extension = extension_key equal extension_value %/extension_eof;
-        extensions = " "* extension (" " extension)*;
+        extension_value = (space* extension_value_chars_nospace @extension_value_mark)* >extension_value_start $err(extension_err);
+        extension = extension_key equal extension_value;
+        extensions = " "* extension (space* " " extension)* space* %/extension_eof;
```

Comment:
I was surprised that `%/extension_eof` worked for each individual extension, rather than just the final one. I'm definitely no Ragel expert.

Reply:
No, I guess I didn't explain myself properly. extension_eof worked as expected. It's just that it captures a value until the last point saved by `@extension_value_mark` while extension_key saves the previous value up to the start of the separator.

I had to move it to a different place here to adjust for the case where a msg is padded and EOF is never reached by extension_value due to it not accepting trailing spaces.